src/components/WwwFrame/PromotionalBanner/Banners/DepositIncentiveBanner.vue

```diff
@@ -46,31 +49,38 @@ export default {
 		return {
 			hasCampaignReward: false,
 			amountToLend: 0,
-			isLoggedin: false,
+			isLoggedIn: false,
 			basketTotal: 0,
 		};
 	},
 	inject: ['apollo', 'cookieStore'],
 	apollo: {
-		query: amountToLendQuery,
-		preFetch: true,
-		variables: {
-			campaignId: '04786358-043c-4c09-af50-2d5e79ceeacd'
-		},
-		result({ data }) {
-			this.amountToLend = parseFloat(data?.my?.depositIncentiveAmountToLend) ?? 0;
-			this.isLoggedin = !!data?.my?.id ?? false;
-			this.basketTotal = parseFloat(data.shop?.basket?.totals?.loanReservationTotal ?? 0);
-			this.hasCampaignReward = !!data?.my?.userAccount?.hasCampaignReward ?? false;
+		preFetch(_config, client) {
+			return client.query({ query: configSettingQuery, variables: { key } }).then(result => {
+				const campaignId = JSON.parse(result?.data?.general?.configSetting?.value ?? '""');
+				return campaignId ? client.query({ query: amountToLendQuery, variables: { campaignId } }) : null;
+			});
 		},
 	},
+	created() {
+		const campaignId = JSON.parse(this.apollo.readQuery({ query: configSettingQuery, variables: { key } })
+			?.general?.configSetting?.value ?? '""');
+		const data = this.apollo.readQuery({
+			query: amountToLendQuery,
+			variables: { campaignId, basketId: this.cookieStore.get('kvbskt') }
+		});
```

Comment:
Should we try/catch these reads, incase the setting is missing or something fails...

Reply:
Oh right, `readQuery` doesn't gracefully fail. Fixed.